crates/ty_python_semantic/src/types/call/arguments.rs

```diff
@@ -122,3 +197,138 @@ impl<'a> DerefMut for CallArgumentTypes<'a, '_> {
         &mut self.arguments
     }
 }
+
+/// Expands a type into its possible subtypes, if applicable.
+///
+/// Returns [`None`] if the type cannot be expanded.
+fn expand_type<'db>(db: &'db dyn Db, ty: Type<'db>) -> Option<Vec<Type<'db>>> {
+    // TODO: Expand enums to their variants
+    match ty {
+        Type::NominalInstance(instance) if instance.class.is_known(db, KnownClass::Bool) => {
+            Some(vec![
+                Type::BooleanLiteral(true),
+                Type::BooleanLiteral(false),
+            ])
+        }
+        Type::Tuple(tuple) => {
+            // Note: This should only account for tuples of known length, i.e., `tuple[bool, ...]`
+            // should not be expanded here.
+            let expanded = tuple
+                .iter(db)
+                .map(|element| {
+                    if let Some(expanded) = expand_type(db, element) {
+                        Either::Left(expanded.into_iter())
+                    } else {
+                        Either::Right(std::iter::once(element))
```

Comment:
Same here

Reply:
This actually doesn't seem to be possible because I'm using `into_iter` here, I can't use `expanded.iter()` because `expanded` is created in this closure.